src/components/views/read/cvForm/WorkExperience.tsx

```diff
@@ -96,59 +99,75 @@ const WorkExperience: FC<Props> = ({ nextStep, prevStep }) => {
     }));
   };
 
-  const handleDone = (cardId: string): void => {
-    const workInfoCard = workCards?.find((card) => card.id === cardId);
-    if (workInfoCard) {
-      let isValid = true;
-      const updatedErrors = { ...errors };
+  const suite = create((data) => {
+    // Validation rules for each field
+    test('jobTitle', 'Job Title is required', () => {
+      enforce(data.jobTitle).isNotEmpty();
+    });
+    test('jobTitle', 'Job Title must be at most 50 characters long', () => {
+      enforce(data.jobTitle).shorterThan(50);
+    });
 
-      if (!workInfoCard.data.jobTitle.trim()) {
-        updatedErrors[`${cardId}-jobTitle`] = 'Job Title is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${cardId}-jobTitle`] = '';
-      }
+    test('institutionName', 'Institution Name is required', () => {
+      enforce(data.institutionName).isNotEmpty();
+    });
 
-      if (!workInfoCard.data.institutionName.trim()) {
-        updatedErrors[`${cardId}-institutionName`] =
-          'Institution Name is required';
-        isValid = false;
-      } else {
-        updatedErrors[`${cardId}-institutionName`] = '';
-      }
+    test(
+      'institutionName',
+      'Institution Name must be at most 30 characters long',
+      () => {
+        enforce(data.institutionName).shorterThan(30);
```

Comment:
Not sure you want to enforce a limit on the institution name length ... [EPFL in french](https://www.lausanne.ch/portrait/savoir-et-formation/hautes-ecoles/hautes-ecoles-liste/epfl.html) is 40 chars long. Even [MIT](https://fr.wikipedia.org/wiki/Massachusetts_Institute_of_Technology) is 37 chars.

Reply:
According to a demo show for Denis and Isabelle, limiting some text fields like first name or last name something like that would be good to not letting the user input a non-sense text.

Reply:
okay !